submariner-operator/submariner-metrics-redesign.md

```diff
@@ -0,0 +1,146 @@
+# Submariner metrics redesign
+
+Related Issue:
+[Exporting Submariner metrics without needing to open ports](https://github.com/submariner-io/submariner/issues/1944)
+
+## Summary
+
+Submariner Gateway and Globanet pods run with HostNetworking. To access metrics exported by these two we need to open ports
+in the infra firewall. Currently design has a limitation where these ports might be in use by some other service,
+causing a conflict and metrics becoming inaccessible. With the move towards minimizing changes needed in cloud infra for
+submariner, there is a use case to export metrics in a way that doesn't require any of these ports to be opened.
+
+## Proposal
+
+This enhancement proposes a redesign of existing metrics for Submariner Gateway and Globalnet by adding a new metrics proxy
+service. Pods running on the same node can access ports without requiring pods to be opened in the cloud infra. Any queries
+for these metrics will land on the proxy which will always run on the same node.
+
+Note that this proposal doesn't address the issue where metrics port is unavailable on the node. That should be addressed as
+a separate enhancement.
+
+## Design Details
+
+* A new `status.submariner.io/gateway=active` label will be added to gateway node that is currently active.
+* Submarine RouteAgent will set the label on the relevant Node.
+* A new DaemonSet `submariner-metrics` will be created. It will use the NodeSelector `status.submariner.io/gateway=active`
+to make sure it always runs on the active gateway node.
+* This DaemonSet will use `schedulerName: default-scheduler` to make sure that metrics proxy pod moves with the active gateway
+label.
+* DaemonSet will also use toleration to make sure it can always be scheduled on gateway node, irrespective of any taints.
+* `submariner-metrics` Pod will use `netcat` to act as proxy and forward all incoming requests on metrics port to the
+metrics port on the node IP.
+* The proxy pod will run two containers if globlanet is enabled, one each for gateway and globalnet.
+* Existing metrics service for gateway and globalnet will be modified to use the new metrics proxy pod. This will be
+done by using same label selector as metrics DaemonSet.
+
+### Submariner Metrics DaemonSet
+
+```yaml
+---
+apiVersion: apps/v1
+kind: DaemonSet
+metadata:
+  name: submariner-metrics
```

Comment:
```suggestion
  name: submariner-metrics-proxy
```

Should probably suffix with `proxy` to make sure its not confused with the metrics services themselves